baseplate/context/thrift.py

```diff
@@ -104,6 +104,12 @@ def _call_thrift_method(self, name, *args, **kwargs):
                             prot.trans.set_header("Sampled", sampled)
                         if span.flags:
                             prot.trans.set_header("Flags", str(span.flags))
+                        if hasattr(span.context, "authentication") \
+                                and span.context.authentication.token:
+                            prot.trans.set_header(
+                                "Authentication",
+                                str(span.context.authentication)
```

Comment:
It feels a bit weird to explicitly check if `.token` is available and then use `__str__` here. Maybe do one or the other? (I lean towards just using `.token` explicitly both places)